### PR TITLE
refactor(crosswalk_traffic_light_estimator): rework parameters

### PR DESCRIPTION
**Pull request by yuntianyi-chen**
## Description

Refactor the ROS Node configuration layout for the crosswalk_traffic_light_estimator package according to the [guideline](https://github.com/orgs/autowarefoundation/discussions/3433).
- Create the schema and parameter files.
- Remove the default values from the source code to avoid duplication.
- Add a parameter `last_detect_color_hold_time` in the README.md file.

## Related links

None

## Tests performed

The refactored node was successfully built and tested by launching this node individually. 

The running result is shown below:
cross

## Notes for reviewers

None

## Interface changes

- Create two configuration files `perception/crosswalk_traffic_light_estimator/config/crosswalk_traffic_light_estimator.param.yaml` and `perception/crosswalk_traffic_light_estimator/schema/crosswalk_traffic_light_estimator.schema.json`
- Parameter values are now required to be passed to the node when launched.

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/crosswalk_traffic_light_estimator/schema/crosswalk_traffic_light_estimator.schema.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "definitions": {
+    "crosswalk_traffic_light_estimator": {
+      "type": "object",
+      "properties": {
+        "use_last_detect_color": {
+          "type": "boolean",
+          "default": true,
+          "description": "If this parameter is true, this module estimates pedestrian's traffic signal as RED not only when vehicle's traffic signal is detected as GREEN/AMBER but also when detection results change GREEN/AMBER to UNKNOWN. (If detection results change RED or AMBER to UNKNOWN, this module estimates pedestrian's traffic signal as UNKNOWN.) If this parameter is false, this module use only latest detection results for estimation. (Only when the detection result is GREEN/AMBER, this module estimates pedestrian's traffic signal as RED.)."
+        },
+        "last_detect_color_hold_time": {
```
Comment (ambroise-arm) [inline perception/crosswalk_traffic_light_estimator/schema/crosswalk_traffic_light_estimator.schema.json:14]: I think we can add bounds for this parameter (>0).

Comment (yuntianyi-chen) [reply perception/crosswalk_traffic_light_estimator/schema/crosswalk_traffic_light_estimator.schema.json:14]: Agreed.

**Review by ambroise-arm — APPROVED** [review]
LGTM

**Review by satoshi-ota — APPROVED** [review]
@ambroise-arm LGTM. Thanks :+1:

**Review by satoshi-ota — CHANGES_REQUESTED** [review]
Oh. Sorry. I think it needs to add parameter config path in launcher file since default parameters are removed.

**Comment by yuntianyi-chen** [comment]
> Oh. Sorry. I think it needs to add parameter config path in launcher file since default parameters are removed.

I think it's already added in the launch.xml file.

**Comment by satoshi-ota** [comment]
> > Oh. Sorry. I think it needs to add parameter config path in launcher file since default parameters are removed.
> 
> I think it's already added in the launch.xml file. `<arg name="crosswalk_traffic_light_estimator_param_file" default="$(find-pkg-share crosswalk_traffic_light_estimator)/config/crosswalk_traffic_light_estimator.param.yaml"/>`

Sorry I missed it. LGTM.

**Review by satoshi-ota — APPROVED** [review]

**Comment by satoshi-ota** [comment]
hmm... It seems that this node called in tier4_perception_launch.launch.xml directly. (not via the one of this package.)

https://github.com/autowarefoundation/autoware.universe/blob/b09f3096f6e9af7d808174e80c2b492f6a5306ec/launch/tier4_perception_launch/launch/traffic_light_recognition/traffic_light.launch.xml#L140-L147

**Comment by miursh** [comment]
@yuntianyi-chen 
Can I add some modification to tier4_perception_launch.launch.xml to launch correctly?

**Comment by yuntianyi-chen** [comment]
> @yuntianyi-chen Can I add some modification to tier4_perception_launch.launch.xml to launch correctly?

@miursh Sure! Please do it.

**Comment by miursh** [comment]
@yuntianyi-chen Done. 5cb0f3f408048f869a5c60148e90083a4ae696e1

**Review by miursh — APPROVED** [review]
LGTM

**Pull request merged**